pkg/collectors/pod.go

```diff
@@ -429,35 +418,84 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
+					c.Name, nodeName)
+			}
+			if mem, ok := req[v1.ResourceMemory]; ok {
+				addGauge(descPodContainerResourceRequestsMemoryBytes, float64(mem.Value()),
+					c.Name, nodeName)
+			}
 
-		if gpu, ok := req[v1.ResourceNvidiaGPU]; ok {
-			addGauge(descPodContainerResourceRequestsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
-		}
+			if gpu, ok := req[v1.ResourceNvidiaGPU]; ok {
+				addGauge(descPodContainerResourceRequestsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
+			}
+
+			if cpu, ok := lim[v1.ResourceCPU]; ok {
+				addGauge(descPodContainerResourceLimitsCpuCores, float64(cpu.MilliValue())/1000,
+					c.Name, nodeName)
+			}
+
+			if mem, ok := lim[v1.ResourceMemory]; ok {
+				addGauge(descPodContainerResourceLimitsMemoryBytes, float64(mem.Value()),
+					c.Name, nodeName)
+			}
 
-		if cpu, ok := lim[v1.ResourceCPU]; ok {
-			addGauge(descPodContainerResourceLimitsCpuCores, float64(cpu.MilliValue())/1000,
-				c.Name, nodeName)
+			if gpu, ok := lim[v1.ResourceNvidiaGPU]; ok {
+				addGauge(descPodContainerResourceLimitsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
+			}
 		}
+	}
+
+	for _, c := range p.Spec.Containers {
+		req := c.Resources.Requests
+		lim := c.Resources.Limits
 
-		if mem, ok := lim[v1.ResourceMemory]; ok {
-			addGauge(descPodContainerResourceLimitsMemoryBytes, float64(mem.Value()),
-				c.Name, nodeName)
+		for resourceName, val := range req {
+			switch resourceName {
+			case v1.ResourceCPU:
+				addGauge(descPodContainerResourceRequests, float64(val.MilliValue())/1000,
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "core")
+			case v1.ResourceStorage:
+				fallthrough
+			case v1.ResourceEphemeralStorage:
+				fallthrough
+			case v1.ResourceMemory:
+				addGauge(descPodContainerResourceRequests, float64(val.Value()),
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "byte")
+			case v1.ResourceNvidiaGPU:
+				addGauge(descPodContainerResourceRequests, float64(val.Value()),
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "device")
+			default:
+				addGauge(descPodContainerResourceRequests, float64(val.Value()),
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "device")
+			}
 		}
 
-		if gpu, ok := lim[v1.ResourceNvidiaGPU]; ok {
-			addGauge(descPodContainerResourceLimitsNvidiaGPUDevices, float64(gpu.Value()), c.Name, nodeName)
+		for resourceName, val := range lim {
+			switch resourceName {
+			case v1.ResourceCPU:
+				addGauge(descPodContainerResourceLimits, float64(val.MilliValue())/1000,
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "core")
+			case v1.ResourceStorage:
+				fallthrough
+			case v1.ResourceEphemeralStorage:
+				fallthrough
+			case v1.ResourceMemory:
+				addGauge(descPodContainerResourceLimits, float64(val.Value()),
+					c.Name, nodeName, sanitizeLabelName(string(resourceName)), "byte")
+			case v1.ResourceNvidiaGPU:
```

Comment:
This resource is now deprecated.

Reply:
Yes. We have not update the client-go and we have already make an agreement on deprecating this once Kubernetes 1.11 is released and the client-go dependency is updated.